src/main/java/core/basesyntax/service/interfaces/TransactionParser.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax.service.interfaces;
+
+import core.basesyntax.dto.FruitTransactionDto;
+import java.util.List;
+
+public interface TransactionParser<T> {
+    List<FruitTransactionDto> parse(List<String> rawString);
```

Comment:
It'll become a Fuit transaction in you implementation if you will pass `FruitTransaction` as `T` in the `<>` operator
```suggestion
    List<T> parse(List<String> rawString);
```